lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -47,7 +47,38 @@ Elixir makes a clear distinction between **documentation** and code comments. Th
 
 ## Long parameter list
 
-TODO.
+#### Problem
+
+In a functional language like Elixir, functions tend to be pure and therefore do not access or manipulate values outside their scopes. Since the input values of these functions are just their parameters, it is natural to expect that functions with a long list of parameters be created to keep them pure. However, when this list has **more than three or four parameters**, the function's interface becomes confusing and prone to errors during use.
+
+#### Example
+
+In the following example, the `loan/6` function has an unnecessarily long list of parameters, causing its interface to be confusing and potentially leading developers to introduce errors during calls to this function.
+
+```elixir
+defmodule Library do
+  # Too many parameters that can be grouped!
+  def loan(user_name, email, password, user_alias, book_title, book_ed) do
+    ...
+  end
+end
+```
+
+#### Refactoring
+
+To eliminate this anti-pattern, related parameters can be grouped using `Maps`, `Structs`, or even `Tuples`. This effectively reduces the number of function parameters, simplifying its interface. In the case of the `loan/6`, its parameters were grouped into two different `Maps`, thereby reducing its arity to `loan/2`:
```

Comment:
Tuples are, IMO, a big no-no when doing this. They're essentially the same thing a parameters: unnamed elements. They're just grouped in a container, but that doesn't help much with this smell IMO. Can we just mention structs and maps?

```suggestion
To address this anti-pattern, related parameters can be grouped using maps or structs. This effectively reduces the number of function parameters, simplifying the function's interface. In the case of `loan/6`, its parameters were grouped into two different maps, thereby reducing its arity to `loan/2`:
```

Reply:
I agree with you that tuples can be less interesting than maps and structs, but I believe there is no need to remove the mention, after all they can be used to decrease the size of the function's parameter list, which is the problem to be treated.  

Out of curiosity, there are some articles about refactoring in Haskell and Erlang, for example, which indicate the use of tuples for this purpose.

Reply:
I think it depends if the tuple is simply created at the call site and immediately unpacked in the function (no benefit), or if it is used to be passed around to subsequent calls (in which case it can be useful).